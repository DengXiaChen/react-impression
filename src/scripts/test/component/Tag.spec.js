import React from 'react';
import { shallow } from 'enzyme';
import { Tag } from '../../components/impression';

const setup = () => {
    const props = {
        theme: 'danger',
        shape: 'pill',
        outline: true,
        closable: true,
        onClose: jest.fn(),
    };

    const wrapper = shallow(
        <Tag {...props} />
    );

    return {
        props,
        wrapper,
    };
};

describe('Button', () => {
    const { props, wrapper } = setup();

    it('render', () => {
        expect(wrapper.find('.tag').exists()).toBe(true);
    });

    it('theme outline', () => {
        expect(wrapper.find('.tag-outline-danger').exists()).toBe(true);
    });

    it('shape', () => {
        expect(wrapper.find('.tag-pill').exists()).toBe(true);
    });

    it('closable', () => {
        expect(wrapper.find('.tag-close').exists()).toBe(true);
    });

    it('onClose', () => {
        wrapper.find('.tag-close').simulate('click');
        expect(props.onClose).toHaveBeenCalled();
    });

    it('ui', () => {
        expect(wrapper).toMatchSnapshot();
    });
});
