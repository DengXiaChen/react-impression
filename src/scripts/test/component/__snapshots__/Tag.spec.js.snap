// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button ui 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <span
    className="tag tag-outline-danger tag-pill"
>
    <span
        className="tag-close"
        onClick={[Function]}
    >
        ×
    </span>
</span>,
  "nodes": Array [
    <span
      className="tag tag-outline-danger tag-pill"
>
      <span
            className="tag-close"
            onClick={[Function]}
      >
            ×
      </span>
</span>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 1,
      "_context": Object {},
      "_currentElement": <Tag
        closable={true}
        onClose={[Function]}
        outline={true}
        shape="pill"
        theme="danger"
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Tag {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "closable": true,
          "onClose": [Function],
          "outline": true,
          "shape": "pill",
          "theme": "danger",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <span
          className="tag tag-outline-danger tag-pill"
>
          <span
                    className="tag-close"
                    onClick={[Function]}
          >
                    ×
          </span>
</span>,
        "_debugID": 2,
        "_renderedOutput": <span
          className="tag tag-outline-danger tag-pill"
>
          <span
                    className="tag-close"
                    onClick={[Function]}
          >
                    ×
          </span>
</span>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Tag
    closable={true}
    onClose={[Function]}
    outline={true}
    shape="pill"
    theme="danger"
/>,
}
`;
